src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventsConsumer.java

```diff
@@ -64,4 +65,8 @@ protected void consumeMessage(ServiceBusSessionReceiverClient sessionReceiver) {
             log.error("Error occurred while closing the session", ex);
         }
     }
+
+    public void stop() {
+        keepRun = false;
```

Comment:
So just for my understanding I guess this is just for testing when you want to cleanup?  I.e. stop the consumption and then cleanup any active threads etc.

Reply:
Did you add a unit test for this?

Reply:
Added unit tests to cover this method. I recon we need this to shutdown executor services gracefully when cloasing the application running.